### ISSUE TITLE
Add ability to connect to device to update information

### ISSUE THREAD

**Issue by bdrakester** [bdrakester/netadmintool#7]
Could be ssh/netmiko, SNMP or API if the devices support it

This functionality could be added to the API.  A PUT request to the device, but instead of the value being directly specified in the JSON, the value for the attribute could be a code indicating to query the device. 

For example, to update the sw_version the JSON in the PUT request would be {'sw_version':'query device'}

JSON supports the NULL datatype, could use that. Makes sense, saying the request says to update that attribute, but with no value specified.

**Comment by bdrakester** [comment]
Should make a new database table for device type, which includes at least make, model, and type used to determine which API/ netmiko connections to use.
Make and model could be just one column.  If that was so, how would I query by Make?

**Comment by bdrakester** [comment]
Can now update the SW Version and Serial number from Cisco ASAs using the API.  Not sure what other attributes would make sense to auto-update.  
Next, want to add similar abilities to Cisco IOS devices.

**Comment by bdrakester** [comment]
Working on adding the update from device to web application, figuring out the best way to ask for credentials for device.  I'd prefer a modal, had it working with JavaScript creating the update button to submit form or show modal for sw_version and serial_number.  I noticed that manual updates for sw_version and serial_number would not work, since it'll display the modal.  Started adding more JavaScript to change the update button back if text input detected using oninput, however that seemed like it would be called over and over, and then if user deleted text it wouldn't revert back.  Next approach was adding onsubmit=Update() to the form instead of modifying the update button based on attribute selected.   Then inside Update() fucntion, check the attribute , etc.. and show modal. However I couldn't show the modal via JavaScript, getting "Uncaught TypeError: ... modal is not a function" error in console.  Then I realized doing this might submit the form twice, since after onsubmit() if submitss the form.  Not sure which approach to go with, or abandon using modal to collect device credentials this way.  
Another way could be adding "Update from device" button if the user selects sw_version or serial_nubmer.
